### PR TITLE
Add an abort routine to Caldav.Xhr

### PR DESCRIPTION
**Pull request by lambdabaa**
@lightsofapollo This is a large diff relative to what was actually changed hehe. In summation:
1. I added an abort method on Caldav.Xhr. It simply calls abort on its xhr request if it's waiting.
2. I changed all of the send methods that make xhr requests and accept callbacks to return the Caldav.Xhr object so that the caller can call abort on it.

This, in theory, should help us back in gaia land :).

### CODE REVIEW

**Comment by lambdabaa** [comment]
I'll add a test for abort later too.

File: test-agent/config.json [hunk 1]
```diff
@@ -1,3 +1,21 @@
```
Comment (lightsofapollo) [inline test-agent/config.json:-1]: we could probably just remove or git ignore this file... it is autogenerated but generally I don't actually use the test-agent UI

Comment (lambdabaa) [inline test-agent/config.json:-1]: Okay sounds good to me.

**Comment by lightsofapollo** [comment]
r+ with abort test... We also need to do some magic for requests (which is what we actually operate on in gaia)

**Comment by lightsofapollo** [comment]
Err the We also X means in a later PR

**Comment by lambdabaa** [comment]
Thanks @lightsofapollo! I'm also not sure offhand about the magic you're talking about so if you could explain to me later this week (I'm off today for Alison's birthday : ), that'd be great.

**Pull request merged**